linalg_ops/matmul/generate_e2e_matmul_tests.py

```diff
@@ -107,57 +97,35 @@ def get_test_shapes(shapes_id: ShapesId):
         ]
     if shapes_id == ShapesId.LARGE:
         return [
-            # some random large sizes
-            TestShape(m=123, k=456, n=789, accumulate=True),
-            TestShape(m=654, k=321, n=234, accumulate=False),
-            # shapes involving vectors (i.e. most rectangular cases)
-            TestShape(m=1, k=1000, n=1000, accumulate=True),  # large vector*matrix
-            TestShape(m=1000, k=1000, n=1, accumulate=True),  # large matrix*vector
-            TestShape(m=1000, k=1000, n=1, accumulate=False),  # large matrix*vector
-            # Be conservative in adding larger shapes. They can result in
-            # high latency tests. If you have to, consider splitting them
-            # out in a way that constrains the latency impact, e.g. by
-            # running on fewer backends/drivers or with fewer generators
-            # (see get_test_generators).
-        ]
-    if shapes_id == ShapesId.GPU_LARGE_ALIGNED:
-        return [
+            # Large aligned sizes.
             TestShape(m=512, k=128, n=512, accumulate=True),
             TestShape(m=512, k=128, n=512, accumulate=False),
-        ]
-    if shapes_id == ShapesId.GPU_LARGE:
-        return [
-            # unaligned cases.
-            TestShape(m=457, k=330, n=512, accumulate=False),
-            TestShape(m=457, k=330, n=514, accumulate=False),
-            TestShape(m=438, k=330, n=514, accumulate=False),
-            TestShape(m=540, k=332, n=516, accumulate=False),
             TestShape(m=1000, k=4, n=512, accumulate=False),
             TestShape(m=4, k=1000, n=512, accumulate=False),
             TestShape(m=512, k=1000, n=4, accumulate=False),
             TestShape(m=512, k=128, n=500, accumulate=False),
+            # Large unaligned sizes.
+            # TestShape(m=123, k=456, n=789, accumulate=True),  # Failing on Vulkan
+            TestShape(m=457, k=330, n=512, accumulate=False),
+            TestShape(m=457, k=330, n=514, accumulate=False),
+            TestShape(m=438, k=330, n=514, accumulate=False),
+            TestShape(m=540, k=332, n=516, accumulate=False),
+            TestShape(m=654, k=321, n=234, accumulate=False),
             TestShape(m=457, k=160, n=512, accumulate=False),
             TestShape(m=512, k=330, n=512, accumulate=False),
+            # Shapes involving vectors (i.e. most rectangular cases).
+            TestShape(m=1, k=1000, n=1000, accumulate=True),  # large vector*matrix
+            TestShape(m=1000, k=1000, n=1, accumulate=True),  # large matrix*vector
+            TestShape(m=1000, k=1000, n=1, accumulate=False),  # large matrix*vector
```

Comment:
I could see these matvec / vecmat test cases getting their own "shapes" group.

Other ideas:

* small
* large
* vectors
* square
* aligned (all dimensions multiples of 4/8)
* unaligned (odd number dimensions)
* model_unet
* model_resnet
* model_llama

What other groupings would make sense to test and label?

Reply:
Let's think about why exactly do we sometimes want to split test cases into separate "shapes" groups?

I say it's:
- To get low-latency small-shapes tests that we can always run, regardless of high latencies and other problems with bigger shapes.
- To get shapes that are deemed "suitable" for a particular target. Originally I had created only "small" and "large", both containing odd (what people call "unaligned") shapes to exercise the most corner cases. Then early GPU codegen folks came in and added GPU tests but said "hold on, we can't actually support odd shapes on GPU". That was the reason why "gpu_aligned" was introduced. I don't suppose that's current anymore?

I don't know a reason to split vectors into their own separate group.

Reply:
I was leaning more towards a single file per test case, like how https://github.com/nod-ai/rocm-gemm-benchmark/tree/main/kernels/mlir is set up. That has "problem sizes" coming from shapes found in production models: https://github.com/nod-ai/rocm-gemm-benchmark/blob/main/gemmbench/problems.py . The focus there is benchmarking though, not correctness testing.

@benvanik pointed out [here on Discord](https://discord.com/channels/689900678990135345/1270451599231156266/1275672591201337404) that grouping was useful for test suites. A few of his points:
> that's painful and wasteful when there's a logical grouping
the cost of launching the processes, creating thread pools/gpu devices, etc to run a single 4x4 matmul is astronomical
small (smoketests that should run frequently) and large (bigger regression tests that run less frequently) and a division by data type is how this stuff naturally falls out
most exclusions per target will be cut on data type - cpu doesn't run bf16, etc
if anything, I'd want a test runner to batch more - especially with how unreliable the AMD devices are